electron/legendary/games.ts

```diff
@@ -185,22 +185,37 @@ class LegendaryGame extends Game {
   public async update() {
     this.state.status = 'updating'
     const { maxWorkers } = (await GlobalConfig.get().getSettings())
-    const workers = maxWorkers === 0 ? '' : `--max-workers ${maxWorkers}`
+    const workers = maxWorkers === 0 ? '' : ` --max-workers ${maxWorkers}`
     const logPath = `"${heroicGamesConfigPath}${this.appName}.log"`
     const writeLog = isWindows ? `2>&1 > ${logPath}` : `|& tee ${logPath}`
-    const command = `${legendaryBin} update ${this.appName} ${workers} -y ${writeLog}`
-
-    try {
-      return await execAsync(command, execOptions).then((v) => {
-        this.state.status = 'done'
-        return v
+    const command = `update ${this.appName}${workers} -y ${writeLog}`.split(' ')
+
+    return new Promise((res) => {
+      const child = spawn(legendaryBin, command)
+      const progress: InstallProgress = {
+        bytes: '0.00MiB',
+        eta: '00:00:00',
+        percent: '0.00%'
+      }
+      ipcMain.handle('requestGameProgress', async (event, appName) => {
+        child.stdout.once('data', (data) => {
+          const isVerifing = `${data}`.includes('Verification')
+          if (appName === this.appName){
+            if (isVerifing){
+              progress.bytes = `${String(data).split(' ')[2]}MiB`
+              progress.percent = `${data}`.split(' ')[3].split(')')[0].replace('(', '')
+              progress.eta = 'verifying'
+            }
+          }
+        })
+        console.log({progress})
```

Comment:
Maybe also `logInfo` instead `console.log` ?

Reply:
No because logInfo doesnt accept objects or anything other than strings.

Reply:
progress is only strings.
logInfo can be
```typescript
logInfo([progress.bytes, progress.eta, progress.percent]);
```
or if there a short thing like progress.getKeyValues ?
```typescript
logInfo([progress.getKeyValues]);
```

Reply:
Progress is not 'only strings', its an object. The values are strings but showing only the values doesnt make sense, we wouldn't know what is about.
We need to change those log functions to accept other types. Test if its an object and then call console log in between brackets, etc.
But I wont do that now.

Also, I'm not deploying this change anymore on this branch since its bugged and need more investigation.